overlord/snapstate/conflict.go

```diff
@@ -134,6 +170,19 @@ func checkChangeConflictExclusiveKinds(st *state.State, newExclusiveChangeKind,
 				ChangeKind: "create-recovery-system",
 				ChangeID:   chg.ID(),
 			}
+		case "revert-snap", "refresh-snap":
+			// Snapd downgrades are exclusive changes
+			if ignoreChangeID != "" && chg.ID() == ignoreChangeID {
+				continue
+			}
+			if !changeIsSnapdDowngrade(st, chg) {
+				continue
+			}
+			return &ChangeConflictError{
+				Message:    "snapd downgrade in progress, no other changes allowed until this is done",
+				ChangeKind: "snapd-downgrade",
```

Comment:
ChangeKind is meant to be usually a real kind of a change, so we need to think a bit about what goes here

Reply:
I don't understand the suggestion, if look at the other cases they mostly repeat chg.Kind() so ChangeKind: chg.Kind() is what we need here

Reply:
Ah right, I misunderstood your initial comment. That makes sense.